src/context/user.provider.tsx

```diff
@@ -0,0 +1,33 @@
+import React, {useState, useEffect} from "react"
+
+//third party
+import firebase, {onAuthStateChanged} from "firebase/auth"
+
+//contexts
+import {UserContext} from "./user.context"
+
+//utils
+import {auth} from "../utils/firebase/firebase.utils"
+
+type UserProviderProps = {
+	children: React.ReactNode
+}
+
+export const UserProvider: React.FC<UserProviderProps> = ({
+	children,
+}: UserProviderProps) => {
```

Comment:
If you've imported react on the top, you do not need to define the function like this.

```suggestion
export const UserProvider = ({ children }: UserProviderProps) => {
```

Reply:
Done.